pkg/utils/rebaseTodo.go

```diff
@@ -32,3 +38,88 @@ func WriteRebaseTodoFile(fileName string, todos []todo.Todo) error {
 	}
 	return err
 }
+
+func MoveTodoDown(fileName string, sha string, action todo.TodoCommand) error {
+	todos, err := ReadRebaseTodoFile(fileName)
+	if err != nil {
+		return err
+	}
+
+	for i := range todos {
+		t := &todos[i]
+		// Comparing just the sha is not enough; we need to compare both the
+		// action and the sha, as the sha could appear multiple times (e.g. in a
+		// pick and later in a merge)
+		if t.Command == action && equalShas(t.Commit, sha) {
+			// The todos are ordered backwards compared to our model commits, so
+			// actually move the commit _up_ in the todos slice (i.e. towards 0)
+
+			// Skip over any entries that we don't show in lazygit's view (e.g.
+			// label, reset, or comment lines).
+			j := i
+			for ; j > 0 && todos[j-1].Commit == "" && todos[j-1].Command != todo.UpdateRef; j -= 1 {
+			}
+
+			if j == 0 {
+				// We expect callers to guard against this
+				return fmt.Errorf("Trying to move first commit down")
+			}
+
+			// Need to allocate a new slice so that we don't overwrite the
+			// original one while we're picking things from it
+			rearrangedTodos := make([]todo.Todo, 0, len(todos))
+			rearrangedTodos = append(rearrangedTodos, todos[0:j-1]...) // everything up to the target position
+			rearrangedTodos = append(rearrangedTodos, todos[i])        // the commit we are moving
+			rearrangedTodos = append(rearrangedTodos, todos[j-1:i]...) // the commits between src and dest that we skipped above
+			rearrangedTodos = append(rearrangedTodos, todos[i+1:]...)  // the remaining commits
+
+			return WriteRebaseTodoFile(fileName, rearrangedTodos)
+		}
+	}
+
+	// Should never get here
+	return fmt.Errorf("Todo %s not found in git-rebase-todo", sha)
+}
+
+func MoveTodoUp(fileName string, sha string, action todo.TodoCommand) error {
```

Comment:
I get the feeling we can DRY this up given a lot of similarity between the two functions, but I haven't thought very hard about it

Reply:
Yes, see e9999f3aa8f5. 😄